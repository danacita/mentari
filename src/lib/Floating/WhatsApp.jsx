import React from 'react';
import PropTypes from 'prop-types';
import whatsappLogo from './whatsapp-logo.svg';

const propTypes = {
  phoneNumber: PropTypes.string
};

const defaultProps = {
  phoneNumber: null
};

const FloatingWhatsAppButton = ({ phoneNumber }) => (
  <React.Fragment>
    <a
      target="_blank"
      rel="noopener noreferrer"
      href={`https://api.whatsapp.com/send?phone=${phoneNumber}`}
    >
      <img
        src={whatsappLogo}
        alt="FloatingWhatsAppButton"
        style={{
          position: 'fixed',
          bottom: '88px',
          width: '71px',
          right: '12px',
          zIndex: 100
        }}
      />
    </a>
  </React.Fragment>
);

FloatingWhatsAppButton.propTypes = propTypes;
FloatingWhatsAppButton.defaultProps = defaultProps;

export default FloatingWhatsAppButton;
