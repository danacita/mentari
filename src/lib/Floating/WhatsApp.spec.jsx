import React from 'react';
import renderer from 'react-test-renderer';

import WhatsApp from './WhatsApp';

describe('WhatsApp', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<WhatsApp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
