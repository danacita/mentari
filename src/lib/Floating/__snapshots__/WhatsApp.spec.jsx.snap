// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatsApp renders correctly 1`] = `
<a
  href="https://api.whatsapp.com/send?phone=null"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="FloatingWhatsAppButton"
    src="whatsapp-logo.svg"
    style={
      Object {
        "bottom": "88px",
        "position": "fixed",
        "right": "12px",
        "width": "71px",
        "zIndex": 100,
      }
    }
  />
</a>
`;
